hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestAbfsInputStreamReadFooter.java

```diff
@@ -234,15 +244,41 @@ private void seekReadAndTest(final AzureBlobFileSystem fs,
       long expectedBCursor;
       long expectedFCursor;
       if (optimizationOn) {
-        if (actualContentLength <= footerReadBufferSize) {
-          expectedLimit = actualContentLength;
-          expectedBCursor = seekPos + actualLength;
+        if (getConfiguration().getHeadOptimizationForInputStream()) {
```

Comment:
Too many variable changes, can we add comments please

Reply:
Have reduced the git diff and have added comments for better code understanding.